pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -738,63 +740,42 @@ func (volTopology *nodeVolumeTopology) GetNodeTopologyLabels(ctx context.Context
 	log := logger.GetLogger(ctx)
 
 	var err error
-	if volTopology.isCSINodeIdFeatureEnabled && volTopology.clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
```

Comment:
I see here we have removed check for `volTopology.clusterFlavor == cnstypes.CnsClusterFlavorVanilla`.
Some of the code is common for Guest Cluster and Vanilla. We don't want certain code executed on the guest cluster.

Reply:
I have moved that line below -  https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2362/files#diff-38cf2065c11a29c2498d29b2007cfa77bab0ce8124a33d1b6a0200175b5985bbR764

I believe the end result is exactly the same as before with perhaps bits of code moved to different function and improving readability little bit.

Reply:
ok. So far it was indirect dependency. After this change, we should mark it as a direct dependency in the go.mod file.